go/common/subscription/utils.go

```diff
@@ -11,7 +11,7 @@ import (
 // ForwardFromChannels - reads messages from the input channels, and calls the `onMessage` callback.
 // Exits when the unsubscribed flag is true.
 // Must be called as a go routine!
-func ForwardFromChannels[R any](inputChannels []chan R, unsubscribed *atomic.Bool, onMessage func(R) error) {
+func ForwardFromChannels[R any](inputChannels []chan R, unsubscribed *atomic.Bool, onMessage func(R) error, onBackendDisconnect func(), timeoutInterval time.Duration) {
```

Comment:
I'd maybe call the function `onBackendDisconnect` -> `onTimeout` to be clearer, I think that's the only time it's called.

Reply:
It is also called when the backend disconnects via the usual means.

Reply:
Usual means meaning when `unsubscribed` is set to true? Think you need to change a `return` to a `break` then in the for loop

Reply:
this bit here:
```	for unclosedInputChannels > 0 {
		chosen, value, ok := reflect.Select(inputCases)
		if !ok {
			// The chosen channel has been closed, so zero out the channel to disable the case
			inputCases[chosen].Chan = reflect.ValueOf(nil)
			unclosedInputChannels--
			continue
		}
```

I'll add a comment to clarify it